data-model/src/main/java/io/micronaut/data/model/DefaultCursoredPage.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.micronaut.core.annotation.Creator;
+import io.micronaut.core.annotation.ReflectiveAccess;
+import io.micronaut.data.model.Pageable.Cursor;
+import io.micronaut.data.model.Pageable.Mode;
+import io.micronaut.serde.annotation.Serdeable;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Implementation of {@link Page} to return when {@link CursoredPageable} is requested.
+ *
+ * @author Andriy Dmytruk
+ * @since 4.8.0
+ * @param <T> The generic type
+ */
+@Serdeable
+class DefaultCursoredPage<T> extends DefaultPage<T> {
```

Comment:
I created this class which stores the cursors for the cursored pagination. 

But I also realized that we have `Slice`, essentially a `Page` without totalCount. Strange that we now how 2 ways to specify if the total count needs to be requested: when defining the method and when supplying the pageable. But I suppose there is no better solution. 

Should we also add `CursoredPage` interface that could be used as a return type and therefore would need a special interceptor 🤔?

Reply:
Yes, I suppose they have that in their examples: https://github.com/jakartaee/data/blob/a37cac5893c9caf0f23bef43e0860c4d20ef8ece/api/src/main/java/jakarta/data/page/CursoredPage.java#L61

Reply:
Probably that would be the best, considering they also have `Optional<Cursor> getCursor(int i)` only there